code/API_definitions/Application_Endpoint_Discovery_API.yaml

```diff
@@ -0,0 +1,596 @@
+---
+openapi: 3.0.3
+info:
+  title: Application Endpoint Discovery API
+  version: 0.1.0-wip
+  description: |
+    # Connect to the Nearest Application Endpoint
+    ---
+    # Overview
+    The Application Discovery API extends beyond the capabilities of the
+    Simple Edge Discovery API by not only locating the nearest Edge Cloud Zone
+    but also directly linking to the application endpoints within those zones.
+    This crucial enhancement enables immediate connection to the most relevant
+    service endpoint, optimizing the application's interaction with
+    edge computing resources.
+    # Introduction
+    Operators often host multiple Edge Cloud Zones across various territories.
+    Connecting an application directly to the nearest operational service
+    endpoint minimizes latency because data travels the shortest possible route.
+    Unlike the Simple Edge Discovery API, which only provides the zone location,
+    this API calculates and provides the actual endpoints within the nearest
+    zone. This approach ensures that developers can instantly integrate these
+    endpoints into their applications, improving performance and user
+    experience.
+    After identifying the nearest application endpoint through the API,
+    developers can:
+    * Connect their application client directly to the chosen application
+    endpoint, enhancing the responsiveness and reliability of the service.
+    * Deploy new instances if no suitable endpoints are within the identified
```

Comment:
In my opinion the API should not "Deploy new instances" if they are not available. The API should be explicit in what it intends to offer to its customer and that is defined as "Discovery" but not the instantiation in my opinion. Either there should be attribute/parameter which user can use to specify what to do if there is no endpoint available or API should return error.

Reply:
Agree with you. We have added to the error 404 another option: "`EndpointNotFound`" . We agree it should return error. On the other hand, when a developer is aware that there is not any endpoints available (404), they can use the EdgeApplicationManagement to instantiate a new instance for that user.